### DISCUSSION TITLE
NMI Pods getting created with forceNamespace by default

### DISCUSSION THREAD

**Discussion by nitin0881** [Azure/aad-pod-identity#1322] [Q&A]
When we are running the command kubectl apply -f https://raw.githubusercontent.com/Azure/aad-pod-identity/master/deploy/infra/deployment-rbac.yaml -n kube-system the nmi pods are getting created with the forceNamespace argument by default. 

We used the describe command to check the nmi pods and it has this argument . 

How should we fix this . It there any other yaml file which we can use to create the nmi pods?

**Comment by aramase** [comment]
`https://raw.githubusercontent.com/Azure/aad-pod-identity/master/deploy/infra/deployment-rbac.yaml` doesn't have the force namespaced flag defined, so it's unlikely that's the issue. I would recommend checking if you have installed pod identity installed before using other different methods (1) helm (2) AKS add-on. 

force namespaced mode is only used in managed mode by default - https://github.com/Azure/aad-pod-identity/blob/master/deploy/infra/managed-mode-deployment.yaml

**Reply by nitin0881** [reply]
Should I redeploy with this file ? Do I need to delete it first before redeploying

**Reply by nitin0881** [reply]
How should I delete the nmi pods

**Comment by nitin0881** [comment]
I delete the nmi pods by deleting the daemon process "nmi" under kube-system namespace and redeployed using the 
kubectl apply -f https://raw.githubusercontent.com/Azure/aad-pod-identity/master/deploy/infra/deployment-rbac.yaml
but I still need the forceNamespaced argument when I describe the nmi pods

**Comment by nitin0881** [comment]
Args:
      --node=$(NODE_NAME)
      --operation-mode=managed
      --forceNamespaced
      --http-probe-port=8085
      --metadata-header-required=true
      --custom-user-agent='managedBy: aks'

These are the arguments I am seeing

**Reply by nitin0881** [reply]
The forceNamespaced flag is setting when I am using the kube-system namespace
kubectl apply -f https://raw.githubusercontent.com/Azure/aad-pod-identity/master/deploy/infra/deployment-rbac.yaml -n kube-system

**Reply by aramase** [reply]
``` 
--custom-user-agent='managedBy: aks'
```

means you have the [AKS Pod Managed Identity add-on](https://learn.microsoft.com/en-us/azure/aks/use-azure-ad-pod-identity) enabled. You don't need to install the add-on and the components using the yaml from here. Either use the add-on or yaml from here to install pod identity.
